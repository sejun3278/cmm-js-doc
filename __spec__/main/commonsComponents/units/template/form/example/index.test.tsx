import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import _ExampleForm from "src/main/commonsComponents/units/template/form/example";

import {
  modalExampleList,
  modalExampleInitProps,
} from "src/main/mainComponents/modules/modal/example/modal.render.example.data";

const commonsProps = {
  isShow: [false, false],
  openModal: (_: number) => () => {},
  closeModal: (_: number) => () => {},
};

export default describe("Example Template Page", () => {
  // 스냅샷
  test("Example Template Page - Snapshot", () => {
    const { container } = render(
      <RecoilRoot>
        <_ExampleForm
          exampleList={modalExampleList}
          initProps={modalExampleInitProps}
          commonsProps={commonsProps}
        />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });

  // h2 태그와 사용 예시 텍스트 검증
  test("Example Template Page - Check have h2 tag", () => {
    const { container } = render(
      <RecoilRoot>
        <_ExampleForm
          exampleList={modalExampleList}
          initProps={modalExampleInitProps}
          commonsProps={commonsProps}
        />
      </RecoilRoot>
    );
    const h2WrapperEle = container.getElementsByClassName("_subTitleTemplate_");
    const h2Ele = h2WrapperEle[0]?.querySelector("h2");
    // h2 태그 존재 여부 검증
    expect(h2Ele).toBeInTheDocument();

    const h2Text = h2Ele?.innerText;
    // 사용 예시 텍스트 검증
    expect(h2Text).toEqual("📝 사용 예시");
  });
});
