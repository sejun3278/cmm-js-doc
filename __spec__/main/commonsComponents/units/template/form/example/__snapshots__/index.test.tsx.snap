// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Template Page Example Template Page - Snapshot 1`] = `
<div>
  <div
    class="_subTitleTemplate_ _example_template_title_ css-k59gj9"
  >
    <div>
      <h2
        class="_title_ _subTitle_"
      />
    </div>
    <div
      class="_subTitleLine_ css-18klhw4"
    />
    <div
      class="css-1sw7z4p"
    >
      <div
        class="css-bqybxg"
      >
        <div>
          <h3
            class="_title_"
          />
        </div>
        <div
          class="css-1bzs2hq"
        />
      </div>
    </div>
  </div>
</div>
`;
